libexec/rtld-elf/aarch64/rtld_c18n_asm.S

```diff
@@ -334,6 +334,20 @@ PATCH_POINT(tramp_save_caller, cookie, 1b)
 PATCH_POINT(tramp_save_caller, ret_args, 2b)
 PATCH_POINT(tramp_save_caller, target, 3b)
 
+/*
+ * Save the address of the current frame to c29 so that unwinders can locate it.
+ * When transitioning to Restricted mode code, its tag must be cleared.
+ */
+TRAMP(tramp_update_fp_exe)
+	mov	c29, TRUSTED_STACK
+TRAMPEND(tramp_update_fp_exe)
+
+TRAMP(tramp_update_fp_res)
+	mov	x29, TRUSTED_STACK_X
```

Comment:
Is the bare address actually useful? A c18n-unaware unwinder still can't use it directly and needs to learn about trusted stacks, so I question how much this makes sense...

Reply:
The kernel will respect capabilities, because otherwise the kernel unwinder would be a way to bypass capability protection.

Reply:
(pmc_save_user_callchain in sys/dev/hwpmc/hwpmc_arm64_md.c if you want to see for yourself)

Reply:
I see. Perhaps I could make it an option to expose c29 just for performance analysis scenarios? Otherwise we need to teach the kernel about trusted frames, which does not sound good at all.